src/vs/workbench/contrib/terminal/browser/links/terminalWordLinkDetector.ts

```diff
@@ -68,6 +69,22 @@ export class TerminalWordLinkDetector implements ITerminalLinkDetector {
 				},
 				startLine
 			);
+
+			// vscode:// and vscode-...:// protocol links
+			if (word.text.match(/vscode(?:-[a-z]+)?:\/\//)) {
```

Comment:
From my testing this all seems to work as expected. Does the following not suffice? (I don't know that we really need to support stable protocol urls from insiders or vice versa, and I think we should pull from product.json to minimize VS Code-specific references in OSS code as much as possible.)

```ts
	if (word.text.startsWith(`${this.productService.urlProtocol}://`)) {
```

Reply:
Also requesting review from @joaomoreno who I think owns the extension protocol url mechanism.

Reply:
That would mean links only work for the current quality, I wanted to capture vscode, vscode-insiders and vscode-exploration, without hardcoding the names

Reply:
> I wanted to capture vscode, vscode-insiders and vscode-exploration, without hardcoding the names

@Tyriar there is `IProductConfiguration#urlProtocol` for this use-case

Reply:
@jrieken but that just gives the current one like @joyceerhl was saying. Shouldn't we be able to open `vscode://` links from vscode insiders?

Reply:
Oh, didn't read properly. I honestly think that not supporting this isn't a real use-case. Strictly, supporting any kind of vscode://-links is slightly fishy because they are meant to be handled by the OS. Extensions that want to do something should just use command://-links. Tho, I can see how this is maybe different for the terminal

Reply:
Thanks, I guess I'll defer that to real custom protocol support in the terminal which is out of scope of this PR.